### PR TITLE
My first pull request :)

### PR DESCRIPTION
**Pull request by greyslater**
Hello,

I'm totally new to this topic(s)
I found this Terraform Module and trying to find out myself how to use it.
Never heard of Terraform before but step by step I'm sure I can do it :)

What is the best way to get started?
Should I fork the repo and from Terraform create add a VSC Provider (github OAuth) to load it as module or can I create a workflow directly with the repo (I guess no, because that's the way I already tried)

I'm sorry to ask this but could you please help me a little bit

Thank you so much / Greetings from Austria
Chris

### CODE REVIEW

**Comment by adinhodovic** [comment]
Hey Chris!

Thanks for the PR.

To get started with Terraform and this module specifically you'll need to initialize the Cloudflare provider and set the relevant `api keys`, `email` and `api token`. The following should work in a new Terraform environment:

```terraform
terraform {
  required_providers {
    cloudflare = {
      version = "2.14.0"
      source  = "cloudflare/cloudflare"
    }
  }
}

provider "cloudflare" {
  account_id = "my-org-account-id"
  email      = "hodovicadin@gmail.com"
}

module "hodovi_cc_maintenance" {
  source          = "git::git@github.com:placementco/terraform-cloudflare-maintenance.git"
  enabled         = true
  cloudflare_zone = "hodovi.cc"
  pattern         = "hodovi.cc/maintenance/*"
  company_name    = "HoneyLogic"
  email           = "support@honeylogic.io"
  logo_url        = "https://s3.eu-west-1.amazonaws.com/honeylogic.io/media/images/Honeylogic-blue.original.png"
  favicon_url     = "https://s3.eu-west-1.amazonaws.com/honeylogic.io/media/images/Honeylogic_-_icon.original.height-80.png"
}
```

You'll also need to set the API key as for example an API Key `export CLOUDFLARE_API_KEY="my-api-key"`.

Now you can initialize terraform `terraform init` locally and create a plan using `terraform plan`. These instructions work with a local setup. 

As you see above you don't want to fork or clone repositories, you'll want to create a `main` repository and reference various modules from that repository and usually a single directory under a repository represents a workspace(as a simple start). You can see that under the module it refers to your github repository, the fork you created: `source          = "git::git@github.com:placementco/terraform-cloudflare-maintenance.git"`.

If your using Terraform cloud; then yes you will need to add a workspace and use a VCS provider to link Terraform cloud to your Git repository containing the Terraform code for that environment. Also you'll need to set any secret variable in the Terraform cloud UI (under variables I think, I'm not yet a heavy Terraform cloud user.). The same snippet should work.

As for the code change in the PR, I would prefer if you removed the `required_version` changes and keep as is. I don't use or track Terraform <0.13. Then I can merge and release a new version of this module that uses the `cloudflare/cloudflare` provider and you can use that as the source instead.

Thanks!

File: versions.tf [hunk 1]
```diff
@@ -1,8 +1,8 @@
 terraform {
   required_providers {
     cloudflare = {
-      source = "terraform-providers/cloudflare"
+      source = "cloudflare/cloudflare"
+      version = "> 2.0.0"
     }
   }
-  required_version = ">= 0.13"
```
Comment (adinhodovic) [inline versions.tf:7]: Keep this line.

**Review by adinhodovic — CHANGES_REQUESTED** [review]

**Comment by greyslater** [comment]
Hi Adin!

Thanks for the very huge response.
I'll have to work me through all of this tomorrow - that's to much right now and for sure there will be one or another question.

As for the code change I'm sure you saw that this was made by @orban

My intention for (my first) pull request was to get a working source (without forking and/or changing myself)

Best wishes
Chris

**Comment by greyslater** [comment]
Hi Adin!

I tested your module and it still works fine.
For testing purpose there was no need to make a own configuration. Let's say you are not able to generate a configuration in Terraform Cloud but you have options to get your configuration from your local Terraform instance or from a github repository.

So I decided to use your module which is directly loaded from your repository as configuration.
I use it as a "root-module" instead of calling it as a "child-module"

I mentioned that it works fine because the cloudflare provider can still be called with `terraform-providers/cloudflare` although `cloudflare/cloudflare` is the official name now.

BTW: `api_token` and `account_id` is just enough if you generate an `api_token` with minimum rights for Workers

`api_token` - (Optional) The Cloudflare API Token. This can also be specified with the `CLOUDFLARE_API_TOKEN` shell environment variable. This is an alternative to `email`+`api_key`. If both are specified, `api_token` will be used over `email`+`api_key` fields.

`account_id` - (Optional) Configure API client with this account ID, so calls use the account API rather than the (default) user API. This is required for other users in your account to have access to the resources you manage. This can also be specified with the `CLOUDFLARE_ACCOUNT_ID` shell environment variable.

Cheers
Chris

**Comment by adinhodovic** [comment]
> Hi Adin!
> 
> I tested your module and it still works fine.
> For testing purpose there was no need to make a own configuration. Let's say you are not able to generate a configuration in Terraform Cloud but you have options to get your configuration from your local Terraform instance or from a github repository.
> 
> So I decided to use your module which is directly loaded from your repository as configuration.
> I use it as a "root-module" instead of calling it as a "child-module"
> 
> I mentioned that it works fine because the cloudflare provider can still be called with `terraform-providers/cloudflare` although `cloudflare/cloudflare` is the official name now.
> 
> BTW: `api_token` and `account_id` is just enough if you generate an `api_token` with minimum rights for Workers
> 
> `api_token` - (Optional) The Cloudflare API Token. This can also be specified with the `CLOUDFLARE_API_TOKEN` shell environment variable. This is an alternative to `email`+`api_key`. If both are specified, `api_token` will be used over `email`+`api_key` fields.
> 
> `account_id` - (Optional) Configure API client with this account ID, so calls use the account API rather than the (default) user API. This is required for other users in your account to have access to the resources you manage. This can also be specified with the `CLOUDFLARE_ACCOUNT_ID` shell environment variable.
> 
> Cheers
> Chris

Hey Chris,

Glad it worked out for you :) I've merged a PR to change the provider name either way and will be closing this. Thanks for the extra guidelines on the Cloudflare provider variables.

I'll close this PR, feel free to add questions to it though.

Cheers

**Pull request closed without merging**